### PR TITLE
[WIP] Initial Travis Setup

### PR DESCRIPTION
**Pull request by parthsujalshah**
1. Updated the travis file as suggested
2. added separate tests for pre-commit

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+jobs:
+  include:
+    - language: node_js # js testing
+
+      node_js:
+      - "stable" # latest stable node version
```
Comment (shubhank-saxena) [inline .travis.yml:6]: Pinpoint to exact version as being used in this project or else there will be engine clash!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      cache: yarn # using yarn to install dependencies
+
+      install:
+        - yarn
```
Comment (shubhank-saxena) [inline .travis.yml:11]: Not sure that this will work. On CI, you need to fetch and build yarn imo!

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - language: python # pre-commit
+
+      python:
+      - 3.8
+
+      install:
+        - pip install pre-commit
+
+      script:
+        - pre-commit run --all-files
```
Comment (shubhank-saxena) [inline .travis.yml:42]: Leave an extra line below!

File: .travis.yml (hunk 1 above)
Comment (jsparmani) [reply .travis.yml:6]: @parthsujalshah work on this

File: .travis.yml (hunk 2 above)
Comment (jsparmani) [reply .travis.yml:11]: @parthsujalshah check this once

File: .travis.yml (hunk 3 above)
Comment (jsparmani) [reply .travis.yml:42]: Okay

**Comment by shubhank-saxena** [comment]
Why are we needing the settings different for CI?

File: .travis.yml [hunk 4]
```diff
@@ -34,10 +29,14 @@ jobs:
       install:
       - poetry install
 
+      env:
+      - SECRET_KEY="!pyfta##&3@51#%t=c^7%in1p)n7m^d+(e$!au%*#1m7y5bmr#"
```
Comment (shubhank-saxena) [inline .travis.yml:33]: Put them inside Travis dashboard. It will pick up from there

Comment (shubhank-saxena) [reply .travis.yml:33]: @jsparmani

**Comment by jsparmani** [comment]
> Why are we needing the settings different for CI?

Yeah, reverted that back

File: .travis.yml (hunk 4 above)
Comment (jsparmani) [reply .travis.yml:33]: Okay

**Pull request merged**